compaction/task_manager_module.cc

```diff
@@ -329,24 +329,32 @@ tasks::is_abortable compaction_task_impl::is_abortable() const noexcept {
     return tasks::is_abortable{!_parent_id};
 }
 
-static future<bool> maybe_flush_all_tables(sharded<replica::database>& db) {
-    auto interval = db.local().get_compaction_manager().flush_all_tables_before_major();
-    if (interval > 0s) {
+static future<bool> maybe_flush_all_tables(sharded<replica::database>& db, bool aggressive_purge_tombstones) {
+    // flush all tables either if
+    // (a) aggressive_purge_tombstones == true (or)
+    // (b) flush_all_tables_before_major > 0s and the configured seconds have elapsed since last all tables flush
+    if (!aggressive_purge_tombstones) {
+        auto interval = db.local().get_compaction_manager().flush_all_tables_before_major();
+        if (interval <= 0s) {
+            co_return false;
+        }
+
         auto when = db_clock::now() - interval;
-        if (co_await replica::database::get_all_tables_flushed_at(db) <= when) {
-            co_await db.invoke_on_all([&] (replica::database& db) -> future<> {
-                co_await db.flush_all_tables();
-            });
-            co_return true;
+        if (co_await replica::database::get_all_tables_flushed_at(db) > when) {
+            co_return false;
         }
     }
-    co_return false;
+
+    co_await db.invoke_on_all([&] (replica::database& db) -> future<> {
+        co_await db.flush_all_tables();
```

Comment:
Yes, needed for commitlog.

Reply:
Maybe we want to call an API named `flush_commitlog()` (which can just call this) so it's clear what we want.

Reply:
There shpuld also be a preceding step before uahing all tables to "seal" the current commitlog segment and force a new one to be opened, so that any new write during flush will go to the new segment and will not reference any segment held up by memtable(s) at the time the function is called.